### ISSUE TITLE
Isosurface plot with colour function deadlock in MPI

### ISSUE THREAD

**Issue by OKaluza** [underworldcode/underworld2#241]
It seems my new implementation of sampling a field at isosurface vertices to colour an isosurface via python/LavaVu was very poorly designed and completely fails in parallel.

This affects the OzBench2008 model in particular, to get around the issue for now, comment out the IsoSurface plots or remove the colour function argument to just plot with a single colour.

Working on a redesign to fix this but it's complicated so may take a while.

**Comment by jmansour** [comment]
Does it work on the stable branch?  Or was this a post-release change?

**Comment by OKaluza** [comment]
Unfortunately it was a change from quite a while back and this bug fell through the cracks so it fails on stable too

**Comment by jmansour** [comment]
Ok!  no stress, there's a few other minor fixes required, so we'll prob do a 2.3.1 soon.  

And just a reminder that we'll need to do the fix on a branch off stable, as we don't want to fold dev into stable.

**Comment by OKaluza** [comment]
Sure thing, do the tests get run in parallel at all currently? I'm guessing not as this probably would have been caught otherwise.

**Comment by RebeccaFarrington** [comment]
Yes, only serial testing at the moment. 
We’ll have to sort out parallel test by next release I’d suggest :)

**Comment by OKaluza** [comment]
Fixed in e6801709c19b7a07fecc1b7a995d739f4e2cb842
Wasn't as painful to fix as I expected, managed to create a python only solution without rewriting any of the C code but will require some more testing on larger proc counts as I've just tested on 4

**Comment by RebeccaFarrington** [comment]
thanks Owen ;)

I’ll test it with a subduction model on raijin & let you know

**Comment by OKaluza** [comment]
Thanks @RebeccaFarrington , I'll leave this open until you have verified it's working for you too as it's a serious issue.

**Comment by jmansour** [comment]
@RebeccaFarrington, note that this fix is only on the dev branch, and the module uses the stable branch (I think!).

**Comment by OKaluza** [comment]
Yes, if we release a 2.3.1 update hopefully we can get the module updated to include this. It's actually a python only fix so no rebuild even needed but I'm sure there will be other issues to include.
